src/ServiceControl/Operations/AuditImporter.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace ServiceControl.Operations
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.ObjectBuilder;
+    using ServiceControl.MessageAuditing;
+    using ServiceControl.Operations.BodyStorage;
+
+    public class AuditImporterFeature : Feature
+    {
+        public AuditImporterFeature()
+        {
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.Container.ConfigureComponent<AuditImporter>(DependencyLifecycle.SingleInstance);
+        }
+    }
+
+    public class AuditImporter
+    {
+        private readonly IEnrichImportedMessages[] enrichers;
+        private readonly BodyStorageFeature.BodyStorageEnricher bodyStorageEnricher;
+
+        public AuditImporter(IBuilder builder, BodyStorageFeature.BodyStorageEnricher bodyStorageEnricher)
+        {
+            this.bodyStorageEnricher = bodyStorageEnricher;
+            enrichers = builder.BuildAll<IEnrichImportedMessages>().Where(e => e.EnrichAudits).ToArray();
+        }
+
+        public ProcessedMessage ConvertToSaveMessage(TransportMessage message)
+        {
+            var metadata = new Dictionary<string, object>
+            {
+                ["MessageId"] = message.Id,
+                ["MessageIntent"] = message.MessageIntent,
+                ["HeadersForSearching"] = string.Join(" ", message.Headers.Values)
+            };
+
+            foreach (var enricher in enrichers)
+            {
+                enricher.Enrich(message.Headers, metadata);
+            }
+
+            bodyStorageEnricher.StoreAuditMessageBody(
+                message.Body,
+                message.Headers,
+                metadata);
+
+            var auditMessage = new ProcessedMessage(message.Headers, metadata)
+            {
+                // We do this so Raven does not spend time assigning a hilo key
+                Id = $"ProcessedMessages/{Guid.NewGuid()}"
```

Comment:
I wonder if a deterministic ID is better here to prevent duplicate `ProcessedMessage`s in the event of restarts at just the right time (see below)

Reply:
That's a very interesting question. I've just copied the code. I would assume that the indexes on the query side deduplicate the results based on the message ID but I need to check it.

Reply:
If it's copy-paste then I don't think this scope should include changing that behaviour.

Reply:
I did a test by sending the same audit message twice and it seems like we are not deduplicating it in the current version so this does not make it worse. I'll try to create a separate fix for the duplication problem.

Reply:
@WilliamBZA I've raised a separate issue to deal (or not deal) with the duplication problem https://github.com/Particular/ServiceControl/issues/1056

Reply:
:+1: